src/python/pants/engine/internals/graph.py

```diff
@@ -236,6 +249,7 @@ async def resolve_target_parametrizations(
                         name: dict(Parametrize.expand(generator.address, override))
                         for name, override in overrides.items()
                     },
+                    generator_name_explicitly_set=target_adaptor.name_explicitly_set,
```

Comment:
It seems like this should always be able to use `generator.name_explicitly_set`?